py/desisim/templates.py

```diff
@@ -2231,15 +2239,17 @@ def _make_simqso_templates(self, redshift=None, rmagrange=None, seed=None,
                 meta['REDSHIFT'][these] = input_qsometa.z
             else:
                 meta['SEED'][these] = seed
-                meta['REDSHIFT'][these] = redshift[these]
-                
+                if self.restframe:
+                    meta['REDSHIFT'][these] = 0
```

Comment:
even if the output is in restframe a qso at z=3 and at z=1 does not look the same, mainly because of the Lyman alpha forest. Maybe we should keep the redshift information.

Reply:
So you're thinking that you would specify an input redshift distribution?  I can implement this, but if you're building archetypes I worry that -- given how strongly the position of the Lyman break / the forest varies with redshift -- you'll end up with many more archetypes that you'd want.

What if you generated a generic set of Lya forest spectra -- over a range of redshifts (or even on-the-fly?) -- separately from the QSO continuum spectra?

Reply:
ok, no sorry. I though it was the output redshift. Not the input one.